core-kotlin-modules/core-kotlin-concurrency-3/src/test/java/com/baeldung/singlerxjavatocoroutinedeferred/SingleRxJavaToCoroutineDeferredUnitTest.kt

```diff
@@ -0,0 +1,214 @@
+package com.baeldung.singlerxjavatocoroutinedeferred
+
+import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.schedulers.Schedulers
+import kotlinx.coroutines.*
+import kotlinx.coroutines.rx3.await
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.slf4j.LoggerFactory
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+import kotlin.test.assertTrue
+
+
+class SingleRxJavaToCoroutineDeferredUnitTest {
+
+    private val logger = LoggerFactory.getLogger("")
+
+    data class Product(val id: Int, val name: String, val price: Double)
+
+    private val allProducts = listOf(
+        Product(1, "Samsung", 1200.0),
+        Product(2, "Oppo", 800.0),
+        Product(3, "Nokia", 450.0),
+        Product(4, "Lenovo", 550.0),
+        Product(5, "ASUS", 400.0)
+    )
+
+    private fun getFilteredProducts(): Single<List<Product>> {
+        return Single.just(
+            allProducts
+        ).map { products ->
+            products.sortedBy { it.price }.filter { it.price > 500 }
+        }.subscribeOn(Schedulers.io())
+    }
+
+    private suspend fun Deferred<*>.assertOver500AndSorted() {
+        assertTrue(actual = this is Deferred<*>)
```

Comment:
I realised this check is actually redundant because the method signature already says we have a Deferred<*>